### PR TITLE
🔥 Hot-reload Syft nodes in your notebooks

### PR DESCRIPTION
**Pull request by itstauq**
## Description
An alternative approach to #8971 and #8987 

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

**Comment by snwagh** [comment]
This is incredible, beyond amazing!

**Review by snwagh — APPROVED** [review]
Works like a charm, this will change the course of Syft's dev team!

**Comment by itstauq** [comment]
@madhavajay there are instances when we run multiple nodes in the same notebook, but are actively developing on just one node. For example, in [tauquir/enclave-prototype/03-ds-execute-code.ipynb](https://github.com/OpenMined/PySyft/blob/tauquir/enclave-prototype/notebooks/experimental/enclaves/03-ds-execute-code.ipynb) we are usually actively working on the `canada-enclave` node but need the other two domain nodes active as well. For such cases, it makes sense to have a separate `hot_reload: bool` flag in `orchestra.launch` instead of tying the hot reloading functionality with `dev_mode=True`. What are your thoughts?

File: packages/syft/src/syft/node/server.py [hunk 1]
```diff
@@ -35,134 +38,96 @@
...
+    association_request_auto_approval: bool = False
+    background_tasks: bool = False
+
+    model_config = SettingsConfigDict(env_prefix="SYFT_", env_parse_none_str="None")
+
+
+def app_factory() -> FastAPI:
+    settings = AppSettings()
+
+    worker_classes = {
```
Comment (BrendanSchell) [inline packages/syft/src/syft/node/server.py:63]: nit: we define this worker_classes dict a few times in the codebase, can we reuse?

**Review by BrendanSchell — APPROVED** [review]
Great work on this Tauquir! Will save a ton of time in notebook debugging🔥

**Comment by rasswanth-s** [comment]
Tested this functionality by changing 
1. NodeMetadata
2. UserService Fields
3. Client Side Fields
In jupyter lab.
Screenshot 2024-07-03 at 5 13 50 PM

**Review by rasswanth-s — APPROVED** [review]
Let's go 
mcqueen

File: packages/syft/src/syft/node/server.py (hunk 1 above)
Comment (itstauq) [reply packages/syft/src/syft/node/server.py:63]: Will address this in a separate PR

**Pull request merged**